### PR TITLE
Jenkins plugin

### PR DESCRIPTION
**Pull request by freben**
This is just a rebase of https://github.com/spotify/backstage/pull/1802, all credit goes to the actual developers of that feature!

### CODE REVIEW

File: plugins/jenkins/src/pages/BuildWithStepsPage/lib/ActionOutput/ActionOutput.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles({
+  expansionPanelDetails: {
+    padding: 0,
+  },
+  button: {
+    order: -1,
+    marginRight: 0,
+    marginLeft: '-20px',
```
Comment (freben) [inline plugins/jenkins/src/pages/BuildWithStepsPage/lib/ActionOutput/ActionOutput.tsx:33]: prefer to use the theme spacing, e.g. `theme.spacing(-3)` so we get even nice multiples

File: plugins/jenkins/src/pages/BuildWithStepsPage/lib/ActionOutput/ActionOutput.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+export const ActionOutput: FC<{
+  url: string;
+  name: string;
+  className?: string;
+  action: any;
+}> = ({ url, name, className }) => {
+  const classes = useStyles();
+
+  useEffect(() => {}, [url]);
```
Comment (freben) [inline plugins/jenkins/src/pages/BuildWithStepsPage/lib/ActionOutput/ActionOutput.tsx:45]: accidentally left around?

File: plugins/jenkins/src/pages/BuildsPage/lib/CITable/CITable.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      <Link component={RouterLink} to={`/jenkins/job?url=${row.id}`}>
+        {row.buildName}
+      </Link>
+    ),
+  },
+  {
+    title: 'Source',
+    render: (row: Partial<CITableBuildInfo>) => (
+      <>
+        <p>
```
Comment (freben) [inline plugins/jenkins/src/pages/BuildsPage/lib/CITable/CITable.tsx:117]: there's `<Typography paragraph>` in material-ui that gives consistent styles

**Review by freben — COMMENTED** [review]
@timja This is great work, there are some very minor nitpicks but I would prefer that we merge this ASAP and then you or others can iterate if/when you want - agreed?

**Comment by timja** [comment]
Happy to iterate on it later, otherwise later today / tomorrow I should be able to review comments and adjust.

**Comment by freben** [comment]
i had push rights to the original branch - hopping back to that one instead

**Pull request closed without merging**